processing/src/main/java/io/druid/segment/DimensionHandlerUtils.java

```diff
@@ -172,10 +185,56 @@ private static ColumnValueSelector getColumnValueSelectorFromDimensionSpec(
   private static <ColumnSelectorStrategyClass extends ColumnSelectorStrategy> ColumnSelectorStrategyClass makeStrategy(
       ColumnSelectorStrategyFactory<ColumnSelectorStrategyClass> strategyFactory,
       String dimName,
-      ColumnCapabilities capabilities
+      ColumnCapabilities capabilities,
+      boolean hasExFn
   )
   {
-    capabilities = getEffectiveCapabilities(dimName, capabilities);
+    capabilities = getEffectiveCapabilities(dimName, capabilities, hasExFn);
     return strategyFactory.makeColumnSelectorStrategy(capabilities);
   }
+
+  private static final Pattern LONG_PAT = Pattern.compile("[-|+]?\\d+");
+
+  // TopNBinaryFn relies on this returning the same Long object when no conversion is needed
+  public static Long convertObjectToLong(Object valObj)
+  {
+    if (valObj instanceof Long) {
+      return (Long) valObj;
+    } else if (valObj instanceof Number) {
+      return ((Number) valObj).longValue();
+    } else if (valObj instanceof String) {
+      try {
+        String s = ((String) valObj).replace(",", "");
```

Comment:
I took this parsing code from `getLongMetric/getFloatMetric` in `MapBasedRow`initially, I kept the comma removal since these transform functions would be used on String columns with "numeric" data, and figured it would be useful to allow that format

I don't mind removing this if someone is against it though.